src/server.py

```diff
@@ -289,32 +272,41 @@ def terminate_sound_for_sender(self, uuid_, connection, sender, invocation,
                                refcount by 1. If set to True, then the refcount
                                is set to 0.
         """
-        assert not (uuid_ in self.registry.sounds and
-                    uuid_ in self.registry.uuids_by_event_id)
+        events_registry = self.registry.sound_events
+
+        assert not (uuid_or_event_id in self.registry.sounds and
+                    events_registry.has_sound_event_id(uuid_or_event_id))
```

Comment:
These checks are done over and over in this function: I'd split the result of `uuid_or_event_id in self.registry.sounds` and `events_registry.has_sound_event_id(uuid_or_event_id)` into two variables here, and use them throughout the function.